types/color/v3/index.d.ts

```diff
@@ -0,0 +1,131 @@
+import convert = require("color-convert");
+
+type ColorParam = Color | string | ArrayLike<number> | number | { [key: string]: any };
+
+interface Color<T extends ColorParam = ColorParam> {
+    toString(): string;
+    toJSON(): Color<T>;
+    string(places?: number): string;
+    percentString(places?: number): string;
+    array(): number[];
+    object(): { alpha?: number | undefined } & { [key: string]: number };
+    unitArray(): number[];
+    unitObject(): { r: number; g: number; b: number; alpha?: number | undefined };
+    round(places?: number): Color;
+    alpha(): number;
+    alpha(val: number): Color;
+    red(): number;
+    red(val: number): Color;
+    green(): number;
+    green(val: number): Color;
+    blue(): number;
+    blue(val: number): Color;
+    hue(): number;
+    hue(val: number): Color;
+    saturationl(): number;
+    saturationl(val: number): Color;
+    lightness(): number;
+    lightness(val: number): Color;
+    saturationv(): number;
+    saturationv(val: number): Color;
+    value(): number;
+    value(val: number): Color;
+    chroma(): number;
+    chroma(val: number): Color;
+    gray(): number;
+    gray(val: number): Color;
+    white(): number;
+    white(val: number): Color;
+    wblack(): number;
+    wblack(val: number): Color;
+    cyan(): number;
+    cyan(val: number): Color;
+    magenta(): number;
+    magenta(val: number): Color;
+    yellow(): number;
+    yellow(val: number): Color;
+    black(): number;
+    black(val: number): Color;
+    x(): number;
+    x(val: number): Color;
+    y(): number;
+    y(val: number): Color;
+    z(): number;
+    z(val: number): Color;
+    l(): number;
+    l(val: number): Color;
+    a(): number;
+    a(val: number): Color;
+    b(): number;
+    b(val: number): Color;
+    keyword(): string;
+    keyword<V extends string>(val: V): Color<V>;
+    hex(): string;
+    hex<V extends string>(val: V): Color<V>;
+    hexa(): string;
+    hexa<V extends string>(val: V): Color<V>;
```

Comment:
`hexa()` is not supported in color v3 ([It's added in v4.2.0](https://github.com/Qix-/color/releases/tag/4.2.0)).
I think it should be removed but it is breaking changes in @types/color v3.
Thus, I keep it.

Reply:
If introducing a breaking change, we should preserve backward compatibility by creating a new directory `types/color/v3/` with copies of the existing type files. Then, make the necessary changes in `types/color/`, and update the package.json to reflect the new version v4.2.3 for the updated type definitions.

Reply:
@jameswlane 
I didn't introduce breaking changes in this PR.

I created a new directory `types/color/v3/` with copies of the existing type files (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70988/commits/19c2b6809cf41c13a8ae5c97a9f04bec49ff6a5e) and update `.npmignore` and `package.json` for v4 (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70988/commits/a77fe1068aedd7242bce058715aae10e14451d0f and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70988/commits/a2120f5246681b393e064f94354f440a39978a66). 

Are there any problems?

Thanks.

Reply:
After further research, I found that the breaking change for v3 was introduced and approved by the color library developer two years ago. Given this context, the changes look good to me, and I’m approving them.